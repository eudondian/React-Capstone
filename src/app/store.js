import { configureStore } from '@reduxjs/toolkit';

import { cryptoCoinsApi } from '../services/cryptoApi';

export default configureStore({
  reducer: {
    [cryptoCoinsApi.reducerPath]: cryptoCoinsApi.reducer,
  },
});
