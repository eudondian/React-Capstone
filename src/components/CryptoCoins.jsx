import React, { useEffect, useState } from 'react';
import millify from 'millify';
import { Link } from 'react-router-dom';
import {
  Card, Row, Col, Input,
} from 'antd';
import { useGetCryptoCoinsQuery } from '../services/cryptoApi';
import Loader from './Loader';

// eslint-disable-next-line react/prop-types
const CryptoCoins = ({ simplified }) => {
  const displayCount = simplified ? 12 : 100;
  const { data: cryptoList, isFetching } = useGetCryptoCoinsQuery(displayCount);
  const [crypto, setCrypto] = useState([]);
  const [searchCoin, setSearchCoin] = useState('');

  useEffect(() => {
    // eslint-disable-next-line max-len
    const filteredCoins = cryptoList?.data?.coins.filter((item) => item.name.toLowerCase().includes(searchCoin.toLowerCase()));

    setCrypto(filteredCoins);
  }, [cryptoList, searchCoin]);
  if (isFetching) return <Loader />;

  return (
    <>
      <div className="container">
        {!simplified && (
          <div className="search-crypto">
            <Input
              placeholder="Search Cryptocurrency"
              onChange={(e) => setSearchCoin(e.target.value)}
              className="input"
            />
          </div>
        )}
        <Row gutter={[32, 32]} className="crypto-card-group">
          {crypto?.map((currency) => (
            <Col
              xs={12}
              sm={12}
              lg={6}
              className="crypto-card"
              key={currency.id}
            >
              <Link to={`/crypto/${currency.id}`} className="link">
                <Card
                  className="card"
                  title={`${currency.rank}.${currency.name}`}
                  extra={(
                    <img
                      className="crypto-img"
                      src={currency.iconUrl}
                      alt="coin logo"
                    />
                  )}
                  hoverable
                >
                  <p>
                    Price:
                    {' '}
                    {millify(currency.price)}
                  </p>
                  <p>
                    Market Cap:
                    {' '}
                    {millify(currency.marketCap)}
                  </p>
                  <p>
                    Daily Change:
                    {' '}
                    {millify(currency.change)}
                    %
                  </p>
                </Card>
              </Link>
            </Col>
          ))}
        </Row>
      </div>
    </>
  );
};

export default CryptoCoins;
