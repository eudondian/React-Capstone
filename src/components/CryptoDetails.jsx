import React, { useState } from 'react';
import {
  Select, Typography, Row, Col, Avatar, Card,
} from 'antd';
import moment from 'moment';

import { useGetCryptosQuery } from '../services/cryptoApi';
import { useGetCryptoNewsQuery } from '../services/cryptoDetailsApi';
import Loader from './Loader';

const demoImage = 'https://www.bing.com/th?id=OVFT.mpzuVZnv8dwIMRfQGPbOPC&pid=News';

const { Text, Title } = Typography;
const { Option } = Select;

// eslint-disable-next-line react/prop-types
const News = ({ simplified }) => {
  const [newsCategory, setNewsCategory] = useState('Cryptocurrency');
  const { data } = useGetCryptosQuery(100);
  const { data: cryptoNews } = useGetCryptoNewsQuery({
    newsCategory,
    count: simplified ? 6 : 12,
  });

  if (!cryptoNews?.value) return <Loader />;

  return (
    <Row gutter={[24, 24]}>
      {!simplified && (
        <Col span={24}>
          <Select
            showSearch
            className="select-news"
            placeholder="Select a Cyrpto"
            optionFilterProp="children"
            onChange={(value) => setNewsCategory(value)}
            // eslint-disable-next-line max-len
            filterOption={(input, option) => option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0}
          >
            <Option value="Cryptocurrency">Cryptocurrency</Option>

            {data?.data?.coins?.map((coin) => (
              // eslint-disable-next-line react/jsx-key
              <Option value={coin.name}>{coin.name}</Option>
            ))}
          </Select>
        </Col>
      )}

      {cryptoNews.value.map((news, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <Col xs={24} sm={12} lg={6} key={i}>
          <Card hoverable className="news-card">
            <a href={news.url} target="_blank" rel="noreferrer">
              <div className="news-image-container">
                <Title className="news-title" level={4}>
                  {news.name}
                </Title>
                <img
                  style={{ maxWidth: '200px', maxHeight: '100px' }}
                  src={news?.image?.thumbnail?.contentUrl || demoImage}
                  alt="News"
                />
              </div>
              <p>
                {news.description.length > 100
                  ? `${news.description.substring(0, 100)}....`
                  : news.description}
              </p>
              <div className="provider-container">
                <div>
                  <Avatar
                    src={
                      news.provider[0]?.image?.thumbnail?.contentUrl
                      || demoImage
                    }
                    alt="News"
                  />
                  <Text className="provider-name">
                    {news.provider[0]?.name}
                  </Text>
                </div>
                <Text>
                  {moment(news.datePublished).startOf('ss').fromNow()}
                </Text>
              </div>
            </a>
          </Card>
        </Col>
      ))}
    </Row>
  );
};

export default News;
