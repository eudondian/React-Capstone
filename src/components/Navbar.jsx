import React, { useState, useEffect } from 'react';
import { Typography } from 'antd';
import { Link } from 'react-router-dom';

const Navbar = () => {
  const [activeMenu, setActiveMenu] = useState(true);
  const [screenSize, setScreenSize] = useState('');

  useEffect(() => {
    const handleResize = () => setScreenSize(window.innerWidth);

    window.addEventListener('resize', handleResize);

    handleResize();

    return () => window.removeEventListener('resize', handleResize);
  }, []);

  useEffect(() => {
    if (screenSize <= 800) {
      setActiveMenu(false);
    } else {
      setActiveMenu(true);
    }
  }, [screenSize]);

  return (
    <div className="nav-container">
      <div className="logo-con">
        <Typography.Title level={3}>
          <Link to="/">CryptoChart</Link>
        </Typography.Title>
      </div>
      {activeMenu && (
        <nav className="menu">
          <div>
            <Link to="/" className="menu-link">
              Home
            </Link>
          </div>
          <div>
            <Link to="/cryptocurrencies" className="menu-link">
              Cryptocurrencies
            </Link>
          </div>
          <div>
            <Link to="/cryptodetails" className="menu-link">
              Crypto Details
            </Link>
          </div>
        </nav>
      )}
    </div>
  );
};

export default Navbar;
