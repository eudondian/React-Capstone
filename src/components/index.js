export { default as Navbar } from './Navbar';
export { default as Cryptocurrencies } from './CryptoCoins';
