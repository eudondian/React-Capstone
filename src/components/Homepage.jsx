import React from 'react';
import { Typography } from 'antd';
import { Link } from 'react-router-dom';

import Cryptocurrencies from './CryptoCoins';
import CryptoDetails from './CryptoDetails';

const { Title } = Typography;

const Homepage = () => (
  <>
    <div className="container">
      <Title level={3} className="title">
        World Crypto Statistics
      </Title>
      <div className="home-heading-container">
        <Title level={4} className="home-title">
          Top 12 Cryptocurrencies in the World
        </Title>
        <Title className="show-more">
          <Link to="/cryptocurrencies">Show More</Link>
        </Title>
      </div>
      <Cryptocurrencies simplified />
    </div>
    <CryptoDetails simplified />
  </>
);

export default Homepage;
