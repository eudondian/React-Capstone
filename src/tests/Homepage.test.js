import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import { Homepage } from '../components';

test('Homepage displays', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Homepage />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
