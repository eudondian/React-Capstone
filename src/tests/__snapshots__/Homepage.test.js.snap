// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage displays 1`] = `
<div
  className="container"
>
  <h3
    className="ant-typography title"
    onClick={null}
    style={
      Object {
        "WebkitLineClamp": undefined,
      }
    }
  >
    World Crypto Statistics
  </h3>
  <div
    className="home-heading-container"
  >
    <h4
      className="ant-typography home-title"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Top 12 Cryptocurrencies in the World
    </h4>
    <h1
      className="ant-typography show-more"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      <a
        href="/cryptocurrencies"
        onClick={[Function]}
      >
        Show More
      </a>
    </h1>
  </div>
  <div
    className="loader"
  >
    <div
      aria-busy={true}
      aria-live="polite"
      className="ant-spin ant-spin-spinning"
    >
      <span
        className="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</div>
`;
